overlord/snapstate/handlers_test.go

```diff
@@ -33,21 +34,180 @@ import (
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/testutil"
+	userclient "github.com/snapcore/snapd/usersession/client"
 )
 
 type handlersSuite struct {
 	baseHandlerSuite
 }
 
 var _ = Suite(&handlersSuite{})
+var mockNBR func() = nil
 
 func (s *handlersSuite) SetUpTest(c *C) {
 	s.baseHandlerSuite.SetUpTest(c)
+	if mockNBR == nil {
+		mockNBR = snapstate.MockNotifyBeginRefresh()
+	}
```

Comment:
```suggestion
	s.mockNBR = snapstate.MockNotifyBeginRefresh()
```

Reply:
Also add a comment why this is done like this. It's a bit anti-pattern of what we normally do.